### PR TITLE
fix dubbo outlier detection due to not considering LocalConnectionFailure

### PR DESCRIPTION
**Pull request by tmsnan**
Commit Message:fix dubbo outlier detection due to not considering LocalConnectionFailure
Additional Description:
Risk Level: Low
Testing: N/A
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: source/extensions/filters/network/dubbo_proxy/router/router_impl.cc [hunk 1]
```diff
@@ -331,7 +331,7 @@ void Router::UpstreamRequest::onPoolFailure(ConnectionPool::PoolFailureReason re
       reason == ConnectionPool::PoolFailureReason::RemoteConnectionFailure) {
     if (reason == ConnectionPool::PoolFailureReason::Timeout) {
       host->outlierDetector().putResult(Upstream::Outlier::Result::LocalOriginTimeout);
-    } else if (reason == ConnectionPool::PoolFailureReason::RemoteConnectionFailure) {
+    } else {
       host->outlierDetector().putResult(Upstream::Outlier::Result::LocalOriginConnectFailed);
     }
```
Comment (wbpcode) [inline source/extensions/filters/network/dubbo_proxy/router/router_impl.cc:336]: Should we set a `LocalOriginConnectFailed` when the failture reason is `RemoteConnectionFailure`?

Comment (tmsnan) [reply source/extensions/filters/network/dubbo_proxy/router/router_impl.cc:336]: In the outlier detection function, LocalOriginConnectFailed field is used to count connection errors，so i think we can set a LocalOriginConnectFailed when the failture reason is RemoteConnectionFailure or LocalConnectionFailure。

Comment (tmsnan) [reply source/extensions/filters/network/dubbo_proxy/router/router_impl.cc:336]: [https://github.com/envoyproxy/envoy/blob/main/envoy/upstream/outlier_detection.h](url)

**Review by wbpcode — DISMISSED** [review]
~~LGTM. Thanks.~~

File: source/extensions/filters/network/dubbo_proxy/router/router_impl.cc (hunk 1 above)
Comment (wbpcode) [inline source/extensions/filters/network/dubbo_proxy/router/router_impl.cc:336]: Sorry for the delayed review. Could you add a simple test case for this minor change? Thanks.

Comment (tmsnan) [reply source/extensions/filters/network/dubbo_proxy/router/router_impl.cc:336]: done

**Review by wbpcode — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**